packages/components/src/guide/index.js

```diff
@@ -38,6 +41,26 @@ export default function Guide( {
 		}
 	}, [ children ] );
 
+	useEffect( () => {
+		const hasLostFocus = ! guideContainer.current.contains(
+			guideContainer.current.ownerDocument.activeElement
+		);
+
+		// Keeping the focus within the guide when the page changes
+		// prevents the modal from closing and avoids focus loss.
+		if ( ! hasLostFocus ) {
+			return;
+		}
+
+		if ( nextButton.current ) {
```

Comment:
Would it be possible to just focus the container? Starting at the bottom is not really the best situation for UX. Then the user must navigate back to the top. Could also focus on button "Close dialog". The dialog container should be able to focus as long as it has tabindex -1.

Reply:
I was wondering the same but it felt to me that when navigating in a guide, in generally you want to click the buttons "next" "next" "next" then "get started". Or "previous" then "next"....

Reply:
I think we should be consistent. This PR fixes the focus loss but the current "Next" buttons (even without this PR) still keep the focus on the button when changing pages. 

I'm happy to change the behavior but we need to be consistent no matter which buttons are used.

Reply:
@youknowriad On latest trunk when the dialog first gains focus, it is on the Close button. I think it would be good to keep it like that and always start at the top.

Reply:
Oh, I see what you mean. Next/Previous buttons in trunk keep focus at the bottom, not on initial focus. Probably still better to start at the top since that is where initial focus is placed.

Reply:
Ok, I've updated this. The close button is now always focused when you switch pages.